sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -153,11 +171,117 @@ public Context startSpan(String methodName, String databaseId, String endpoint,
+                        ModelBridgeInternal.getPayloadLength(itemResponse), containerId,
+                        databaseId, operationType, resourceType, consistencyLevel,
+                        (float) itemResponse.getRequestCharge());
+                }
+            }).doOnError(throwable -> {
+                if (Configs.isClientTelemetryEnabled(BridgeInternal.isClientTelemetryEnabled(client)) && throwable instanceof CosmosException) {
+                    CosmosException cosmosException = (CosmosException) throwable;
+                    fillClientTelemetry(client, cosmosException.getDiagnostics(), cosmosException.getStatusCode(),
+                        null, containerId,
+                        databaseId, operationType, resourceType, consistencyLevel,
+                        (float) cosmosException.getRequestCharge());
+                }
+            });
+    }
+
     private void end(int statusCode, Throwable throwable, Context context) {
         if (throwable != null) {
             tracer.setAttribute(TracerProvider.ERROR_MSG, throwable.getMessage(), context);
             tracer.setAttribute(TracerProvider.ERROR_TYPE, throwable.getClass().getName(), context);
         }
         tracer.end(statusCode, throwable, context);
     }
+
+    public void fillClientTelemetry(CosmosAsyncClient cosmosAsyncClient,
+                                    CosmosDiagnostics cosmosDiagnostics,
+                                    int statusCode,
+                                    Integer objectSize,
+                                    String containerId,
+                                    String databaseId,
+                                    OperationType operationType,
+                                    ResourceType resourceType,
+                                    ConsistencyLevel consistencyLevel,
+                                    float requestCharge) {
+        ClientTelemetry telemetry = BridgeInternal.getContextClient(cosmosAsyncClient).getClientTelemetry();
+        ReportPayload reportPayloadLatency = createReportPayload(cosmosAsyncClient, cosmosDiagnostics,
+            statusCode, objectSize, containerId, databaseId
+            , operationType, resourceType, consistencyLevel, ClientTelemetry.REQUEST_LATENCY_NAME,
+            ClientTelemetry.REQUEST_LATENCY_UNIT);
+        ConcurrentDoubleHistogram latencyHistogram = telemetry.getClientTelemetryInfo().getOperationInfoMap().get(reportPayloadLatency);
+        if (latencyHistogram != null) {
+            ClientTelemetry.recordValue(latencyHistogram, cosmosDiagnostics.getDuration().toNanos()/1000);
+        } else {
+            if (statusCode == HttpConstants.StatusCodes.OK || statusCode == HttpConstants.StatusCodes.CREATED) {
+                latencyHistogram = new ConcurrentDoubleHistogram(ClientTelemetry.REQUEST_LATENCY_MAX_MICRO_SEC, ClientTelemetry.REQUEST_LATENCY_SUCCESS_PRECISION);
+            } else {
```

Comment:
I think the right groups here are

200 - 299 --> success
400 - 499 --> client_failure
>= 500 --> service failure

Reply:
Discussed offline, changed success scenario to cover status code between 200-299